src/app/actions/teacherActions.ts

```diff
@@ -0,0 +1,68 @@
+'use server'
+
+import { db } from '@/db/db'
+import { TeacherRepository } from '@/repositories/teacherRepository'
+import { TeacherAvailabilityRepository } from '@/repositories/teacherAvailabilityRepository'
+
+const teacherRepository = new TeacherRepository(db)
+const availabilityRepository = new TeacherAvailabilityRepository(db)
+
+export async function createTeacher(data: any) {
+  const { availability: availabilityData, ...teacherData } = data
+  const newTeacher = await teacherRepository.create(teacherData)
+  if (!newTeacher) {
+    throw new Error('Failed to create teacher')
+  }
+  
+  if (availabilityData && availabilityData.length > 0) {
+
+    availabilityRepository.create(availabilityData.map((a: any) => ({ ...a, teacher_id: newTeacher.id })))
```

Comment:
Bitte aussagekräftigere Variablennamen verwenden
```suggestion
    availabilityRepository.create(availabilityData.map((singleAvailability: any) => ({ ...singleAvailability, teacher_id: newTeacher.id })))
```